sdk/eventhub/Azure.Messaging.EventHubs/tests/Producer/EventHubProducerClientTests.cs

```diff
@@ -645,6 +646,550 @@ public async Task CloseAsyncDoesNotCloseTheConnectionWhenNotOwned()
+        /// </summary>
+        ///
+        [Test]
+        public async Task EventHubProducerClientShouldPickAnItemFromPool()
+        {
+            var options = new SendEventOptions { PartitionId = "0" };
+            var transportProducer = new ObservableTransportProducerMock();
+            var eventHubConnection = new MockConnection(() => transportProducer);
+            var retryPolicy = new EventHubProducerClientOptions().RetryOptions.ToRetryPolicy();
+            var mockTransportProducerPool = new MockTransportProducerPool(new ObservableTransportProducerMock(), eventHubConnection, retryPolicy);
+            var mockPooledProducer = mockTransportProducerPool.GetPooledProducer(options.PartitionId) as MockPooledProducer;
+            var producerClient = new EventHubProducerClient(eventHubConnection, transportProducer, mockTransportProducerPool);
+            var events = new EventData[0];
+
+            await producerClient.SendAsync(events, options);
+
+            Assert.That(mockTransportProducerPool.GetPooledProducerWasCalled, Is.True, $"The method { nameof(TransportProducerPool.GetPooledProducer) } should have been called.");
+        }
+
+        /// <summary>
+        ///   Verifies that when calling <see cref="EventHubProducerClient.SendAsync" /> for batches
+        ///   a <see cref="TransportProducer"/> is taken from a <see cref="TransportProducerPool" /> when a partition id is specified.
+        /// </summary>
+        ///
+        [Test]
+        public async Task EventHubProducerClientShouldPickAnItemFromPoolWithABatch()
+        {
+            var batchOptions = new CreateBatchOptions { PartitionId = "0" };
+            var batch = new EventDataBatch(new MockTransportBatch(), batchOptions.ToSendOptions());
+            var transportProducer = new ObservableTransportProducerMock();
+            var eventHubConnection = new MockConnection(() => transportProducer);
+            var retryPolicy = new EventHubProducerClientOptions().RetryOptions.ToRetryPolicy();
+            var mockTransportProducerPool = new MockTransportProducerPool(new ObservableTransportProducerMock(), eventHubConnection, retryPolicy);
+            var mockPooledProducer = mockTransportProducerPool.GetPooledProducer(batchOptions.PartitionId) as MockPooledProducer;
+            var producerClient = new EventHubProducerClient(eventHubConnection, transportProducer, mockTransportProducerPool);
+
+            await producerClient.SendAsync(batch);
+
+            Assert.That(mockTransportProducerPool.GetPooledProducerWasCalled, Is.True, $"The method { nameof(TransportProducerPool.GetPooledProducer) } should have been called (for a batch).");
+        }
+
+        /// <summary>
+        ///   Verifies that a <see cref="TransportProducerPool.PooledProducer.DisposeAsync()" /> is called
+        ///   to unregister a producer from the pool.
+        /// </summary>
+        ///
+        [Test]
+        public async Task EventHubProducerClientShouldCloseAProducer()
```

Comment:
May want to give a bit more context here and mention that we're closing a pooled producer.   I was a bit confused until I dug into the details.  😄